src/components/application_manager/include/application_manager/application_manager_impl.h

```diff
@@ -763,6 +830,9 @@ class ApplicationManagerImpl
 
   void SendMessageToHMI(const commands::MessageSharedPtr message) OVERRIDE;
 
+  void RemoveHMIFakeParameters(
```

Comment:
@yang1070 
"Fake" means absent in API. This method does cleanup to avoid unnecessary copying of data fields that never will be used. After it has been applied object is considered as matching to API and ready to be processed.